src/harness/fourslash.ts

```diff
@@ -507,8 +507,17 @@ namespace FourSlash {
         }
 
         private getAllDiagnostics(): ts.Diagnostic[] {
-            return ts.flatMap(this.languageServiceAdapterHost.getFilenames(), fileName =>
-                ts.isAnySupportedFileExtension(fileName) ? this.getDiagnostics(fileName) : []);
+            return ts.flatMap(this.languageServiceAdapterHost.getFilenames(), fileName => {
+                if (!ts.isAnySupportedFileExtension(fileName)) {
+                    return [];
+                }
+
+                const baseName = ts.getBaseFileName(fileName);
+                if (baseName === "package.json" || baseName === "tsconfig.json" || baseName === "jsconfig.json") {

```

Comment:
> baseName === "tsconfig.json" 

But if we have tsconfig with custom name will it work?

Reply:
have the same problem

Reply:
@sheetalkamat can you please take a look?

Reply:
I think you probably want to raise a new issue for this; I doubt people are looking at this as the original issue was solved and shipped with 3.0.0 AFAIK